src/service/viewer-impl.js

```diff
@@ -1012,8 +1012,13 @@ export class Viewer {
           eventType, data, awaitResponse));
     }
 
+    // If there is no viewer, no further action is needed.
     if (!this.messagingReadyPromise_) {
-      return Promise.reject(getChannelError());
+      if (awaitResponse) {
+        return Promise.reject(getChannelError());
+      } else {
+        return Promise.resolve();
```

Comment:
But if there's no viewer, there's no need to do the action below like adding it into the queue.